### ISSUE TITLE
[DX] Views display machine names and template file overrides.

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#2339]
When you create a block or page from the simplified Views UI screen, the machine name is set to `block` or `page` respectively. However when you create your first block, page, feed, etc. from the main (advanced) Views UI screen, the machine names are appended with a number (e.g. `page_1`).

The basic machine names are problematic when trying to override that display's template file if another display of the same type exists. For example, if I have a block created from the simplified UI screen, it's machine name is `block` and so the most specific template file I can use is `views-view-unformatted--test--block.tpl.php`. However, if I have another block display, it's machine name is `block_1` and I can override it using `views-view-unformatted--test--block-1.tpl.php`. The problem is that the template file override for`block` (`views-view-unformatted--test--block.tpl.php`) applies to `block_1` as well, since it's a block-type too. There's therefore no way to override the template file for `block` without it affecting `block_1` as well.

I therefore propose that the machine names get changed to `[display-type]_1` by default (from the simplified UI screen) to avoid this problem, though I'm not sure what ongoing affects this change might have...

**Comment by klonos** [comment]
An "easy" fix would be to be adding a `0` to the first view displays, so that the override for it would be `views-view-unformatted--test--block-0.tpl.php`. That way, the `views-view-unformatted--test--block.tpl.php` template would be left in place, serving the purpose it's meant to be serving: overriding all block displays.

...as is the norm in life though, "easy" is never easy. What would then happen to websites that are already using `views-view-unformatted--test--block.tpl.php` to override things? Perhaps a simple rename of the template (to add the `0`) is not that big a change, a change though nonetheless. Hmm.

Thank you for filing this issue @BWPanda. Gave me the motivation I needed to file an issue for a related UI problem that I have noticed with views display names but never actually filed an issue for: #2499

**Comment by klonos** [comment]
@quicksketch @serundeputy @herbdool what do you think would be the best approach to tackle this? Will appending `_0` to the page/block displays created by the simplified Views UI page (`admin/structure/views/add`) suffice (resulting in `page_0` and `block_0`)? ...or should we perhaps be generating their machine names based on the values of the "Page title" and "Block title" fields (resulting in something like `page_my_title` and `block_my_title` for machine names)?

**Comment by klonos** [comment]
Agreed re not changing existing views @herbdool, but can you elaborate as to why `_1` and not `_0`?

**Comment by herbdool** [comment]
Because I'm assuming that the regular screen can handle inserting block_2 if the wizard creates block_1 and won't try to create a second block_1. We should test that approach first.

**Comment by BWPanda** [comment]
Here's an easy PR. Wonder if it'll work :wink: https://github.com/backdrop/backdrop/pull/3070

**Comment by herbdool** [comment]
Lots of errors in the tests. The ZenCI links never seem to work for me anymore so @BWPanda might need to run tests locally. Might be something simple like the tests are expecting a different machine name.

**Comment by BWPanda** [comment]
PR updated. See if tests work better now...
